dotCMS/src/main/java/com/dotcms/rest/api/v1/site/SiteResource.java

```diff
@@ -244,4 +297,633 @@ public final Response switchSite(
+
+        if (null == host) {
+            throw new IllegalArgumentException("Site: " + hostId + " does not exists");
+        }
+
+        this.siteHelper.unarchive(host, user, pageMode.respectAnonPerms);
+        return Response.ok(new ResponseEntityView(this.toView(host))).build();
+    }
+
+    /**
+     * Delete a host
+     * Default host can not be deleted
+     * @param httpServletRequest
+     * @param httpServletResponse
+     * @param hostId
+     * @return
+     * @throws DotDataException
+     * @throws DotSecurityException
+     * @throws PortalException
+     * @throws SystemException
+     */
+    @DELETE
+    @Path("/{hostId}")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public void deleteSite(@Context final HttpServletRequest httpServletRequest,
+                                @Context final HttpServletResponse httpServletResponse,
+                                @Suspended final AsyncResponse asyncResponse,
+                                @PathParam("hostId")  final String hostId) throws DotDataException, DotSecurityException, PortalException, SystemException {
+
+        final User user = new WebResource.InitBuilder(this.webResource)
+                .requestAndResponse(httpServletRequest, httpServletResponse)
+                .requiredBackendUser(true)
+                .rejectWhenNoUser(true)
+                .requireLicense(true)
+                .init().getUser();
+
+        Logger.debug(this, ()-> "deleting the site: " + hostId);
+
+        final PageMode      pageMode      = PageMode.get(httpServletRequest);
+        final boolean respectFrontendRoles = pageMode.respectAnonPerms;
+        final Host host = pageMode.respectAnonPerms? this.siteHelper.getSite(user, hostId):
+                this.siteHelper.getSiteNoFrontEndRoles(user, hostId);
+
+        if (null == host) {
+            throw new IllegalArgumentException("Site: " + hostId + " does not exists");
+        }
+
+        if(host.isDefault()) {
+
+            throw new DotStateException("the default site can't be deleted");
+        }
+
+        final Future<Boolean> deleteHostResult = this.siteHelper.delete(host, user, respectFrontendRoles);
+        if (null == deleteHostResult) {
+
+            throw new DotStateException("the Site: " + hostId + " couldn't be deleted");
+        } else {
+
+            try {
+                asyncResponse.resume(new ResponseEntityView(deleteHostResult.get()));
+            } catch (Exception e) {
+                asyncResponse.resume(ResponseUtil.mapExceptionResponse(e));
+            }
+        }
+    }
+
+
+    /**
+     * Make a host as a default
+     * @param httpServletRequest
+     * @param httpServletResponse
+     * @param hostId
+     * @return
+     * @throws DotDataException
+     * @throws DotSecurityException
+     * @throws PortalException
+     * @throws SystemException
+     */
+    @PUT
+    @Path("/_makedefault/{hostId}")
```

Comment:
I think it's better to have the hostId and then the action the exec. `/{hostId}/_makedefault`